lib/test-worker.js

```diff
@@ -34,73 +30,40 @@ sourceMapSupport.install({
 		if (sourceMapCache[source]) {
 			return {
 				url: source,
-				map: sourceMapCache[source]
+				map: fs.readFileSync(sourceMapCache[source], 'utf8')
 			};
 		}
 	}
 });
 
-var requireFromString = require('require-from-string');
 var loudRejection = require('loud-rejection/api')(process);
 var serializeError = require('serialize-error');
 var send = require('./send');
-
-// if generators are not supported, use regenerator
-var options = {
-	sourceMaps: true
-};
+var installPrecompiler = require('require-precompiled');
+var cacheDir = opts.cacheDir;
 
 // check if test files required ava and show error, when they didn't
 exports.avaRequired = false;
 
-// try to load an input source map for the test file, in case the file was
-// already compiled once by the user
-var inputSourceMap = sourceMapSupport.retrieveSourceMap(testPath);
-if (inputSourceMap) {
-	// source-map-support returns the source map as a json-encoded string, but
-	// babel requires an actual object
-	options.inputSourceMap = JSON.parse(inputSourceMap.map);
-}
-
-// include test file
-var cachePath = hasha(cacheKey(testPath));
-var hashPath = cachePath + '_hash';
-
-var prevHash = cache.getSync(hashPath, {encoding: 'utf8'});
-var currHash = hasha.fromFileSync(testPath);
-
-if (prevHash === currHash) {
-	var cached = JSON.parse(cache.getSync(cachePath));
-
-	sourceMapCache[testPath] = cached.map;
-	requireFromString(cached.code, testPath, {
-		appendPaths: module.paths
-	});
-} else {
-	var createEspowerPlugin = require('babel-plugin-espower/create');
-	var babel = require('babel-core');
-
-	// initialize power-assert
-	var powerAssert = createEspowerPlugin(babel, {
-		patterns: require('./enhance-assert').PATTERNS
-	});
-
-	options.presets = [require('babel-preset-stage-2'), require('babel-preset-es2015')];
-	options.plugins = [powerAssert, require('babel-plugin-transform-runtime')];
+installPrecompiler(function (filename) {
+	var precompiled = opts.precompiled[filename];
+	if (precompiled) {
+		sourceMapCache[filename] = path.join(cacheDir, precompiled + '.map');
+		return fs.readFileSync(path.join(cacheDir, precompiled + '.js'), 'utf8');
+	}
+	return null;
+});
 
-	var transpiled = babel.transformFileSync(testPath, options);
+// Modules need to be able to find `babel-runtime`, which is nested in our node_modules w/ npm@2
+var nodeModulesDir = path.join(__dirname, '../node_modules');
+var oldNodeModulesPaths = module.constructor._nodeModulePaths;
+module.constructor._nodeModulePaths = function () {
+	var ret = oldNodeModulesPaths.apply(this, arguments);
+	ret.push(nodeModulesDir);
+	return ret;
+};
```

Comment:
@jamestalmage: Why not just modify `process.env.NODE_PATH` (I think that still works)? Or you could use [`app-module-path`](http://npm.im/app-module-path) and just pass in the path to node modules.

Or, you know what? This is probably fine. But the bikeshed is to be blue.